packages/editor/src/components/post-publish-panel/maybe-upload-media.js

```diff
@@ -25,25 +30,71 @@ function flattenBlocks( blocks ) {
 	return result;
 }
 
-function Image( block ) {
+/**
+ * Determine whether a block has external media.
+ *
+ * Different blocks use different attribute names (and potentially
+ * different logic as well) in determining whether the media is
+ * present, and whether it's external.
+ *
+ * @param {{name: string, attributes: Object}} block The block.
+ * @return {boolean?} Whether the block has external media
+ */
+function hasExternalMedia( block ) {
```

Comment:
just a mental todo, but I'm curious if we should include a new role=media for attributes to leverage stuff like this and the display of little thumbnails on list view. cc @youknowriad @ntsekouras

Reply:
I think it would be a good idea to declaratively know which attributes are media but there's a few things that we need to know:

 - The type of media (video, image, audio)
 - Whether it's saved or not by checking for the existence of `id` set on the block and it seems the attribute name is not consistent.

Reply:
That's something worth exploring for sure, but in order to make some media blocks more semantic, we'd need more than one role. For start we'd need the `id` and the `url`, which could have different names in blocks, like the example below with Image and Media Text.

I'd probably expect to also require a `mediaType` because video and audio have different tag attributes and a preview for example would have to be generated differently. You can see an example of such [handling for media](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inserter/media-tab/utils.js#L17) tab in inserter.

Finally, right now these attributes have the `content` role and that could simple mean that the new `media` roles are also considered `content`. This would require more testing though, to validate that this is the case for all blocks that use them.

Reply:
Let's open an issue so we can track that a bit, it will get lost otherwise. There's a lot of things for which media roles would be useful to know.

Reply:
Opened an issue for this: https://github.com/WordPress/gutenberg/issues/65186